tests/test_config.py

```diff
@@ -175,6 +175,20 @@ def test_cmd_line():
         app = NoConfigApp()
         assert app.cfg.preload_app
 
+
+def test_cmd_line_invalid_setting(capsys):
+    with AltArgs(["prog_name", "-q", "bar"]):
+        with pytest.raises(SystemExit):
+            NoConfigApp()
+        out, err = capsys.readouterr()
+
+        expected = (
+            "usage: no_usage\ngunicorn_test: error: unrecognized "
+            "arguments: -q\n"
+        )
+        assert err == expected
```

Comment:
On a second thought, we can just look for "error: unrecognized arguments: -q" and change the assert statement to

```py
assert  "error: unrecognized arguments: -q" in err
```